daemon/daemon.go

```diff
@@ -250,15 +250,9 @@ func (d *Daemon) Dying() <-chan struct{} {
 	return d.tomb.Dying()
 }
 
-func (d *Daemon) auther(r *http.Request) (store.Authenticator, error) {
+func (d *Daemon) auther(user *auth.UserState) (store.Authenticator, error) {
```

Comment:
Hm, did you have a different idea when you said "We can very easily carry around some sort of device-specific information in an auth.User."? With Authenticator gone, we either need to annotate the returned UserState with DeviceState (or just device credentials), or thread extra arguments through all the Store methods.